File: core/src/backendServer/models/MachineGunProjectile.js
const Projectile = require('./Projectile');
const ProjectileType = require('./ProjectileType');
const constants = require('../settings/constants');

class MachineGunProjectile extends Projectile {
    constructor(x = 0, y = 0, xSpeed = 0, ySpeed = 0, id) {
        super(x, y, xSpeed, ySpeed, constants.STANDARD_PROJECTILE_WIDTH / 2, ProjectileType.MACHINE_GUN,
            constants.MACHINE_GUN_BULLET_SPEED, id);
    }
}

module.exports = MachineGunProjectile;

File: core/src/com/mygdx/game/model/projectile/MachineGunProjectile.kt
package com.mygdx.game.model.projectile

import com.badlogic.gdx.graphics.Texture
import com.mygdx.game.settings.MACHINE_GUN_BULLET_SPEED
import com.mygdx.game.settings.STANDARD_PROJECTILE_WIDTH

class MachineGunProjectile(
        x: Float = 0f,
        y: Float = 0f,
        xSpeed: Float = 0f,
        ySpeed: Float = 0f,
        texture: Texture) : Projectile(
        x,
        y,
        xSpeed,
        ySpeed,
        texture,
        STANDARD_PROJECTILE_WIDTH / 2,
        MACHINE_GUN_BULLET_SPEED)